pkg/kn/commands/service/configuration_edit_flags.go

```diff
@@ -427,6 +432,29 @@ func (p *ConfigurationEditFlags) Apply(
 		servinglib.UpdateUser(template, p.PodSpecFlags.User)
 	}
 
+	if cmd.Flags().Changed("scale-init") {
+		containsAnnotation := func(annotationList []string, annotation string) bool {
+			for _, element := range annotationList {
+				if strings.Contains(element, annotation) {
+					return true
+				}
+			}
+			return false
+		}
+
+		if cmd.Flags().Changed("annotation") && containsAnnotation(p.Annotations, "autoscaling.knative.dev/initialScale") {
+			return fmt.Errorf("only one of the --scale-init or --annotation autoscaling.knative.dev/initialScale can be specified")
```

Comment:
good catch, but you should probably refer to the constant that is used by knative serving for this annotation instead of using it literally here.

Also, I wonder whether we should throw an error here. If so we should do the same for the other scale related annotation(s).

Reply:
Pushed the changes using constant. And about the changes for other scale related issues, we can do it in a separate PR!